searchcore/src/vespa/searchcore/proton/matching/query.cpp

```diff
@@ -126,10 +126,18 @@ Query::reserveHandles(const IRequestContext & requestContext, ISearchContext &co
     LOG(debug, "original blueprint:\n%s\n", _blueprint->asString().c_str());
     if (_whiteListBlueprint) {
         auto andBlueprint = std::make_unique<AndBlueprint>();
-        (*andBlueprint)
-            .addChild(std::move(_blueprint))
-            .addChild(std::move(_whiteListBlueprint));
-        _blueprint = std::move(andBlueprint);
+        if (dynamic_cast<const search::queryeval::RankBlueprint*>(_blueprint.get())) {
```

Comment:
since you always do the dynamic_cast can't you just save in 'rank' variable?
```
        search::queryeval::RankBlueprint * rank = dynamic_cast<search::queryeval::RankBlueprint*>(_blueprint.get());
        if (rank) {
```